### ISSUE TITLE
Can't unignore a file when directory ignored

### ISSUE THREAD

**Issue by abaksha-sc** [eslint/eslint#12981]
**Tell us about your environment**

Node version: v12.14.1
npm version: v6.13.7
Local ESLint version: v6.8.0 (Currently used)
Global ESLint version: Not found

**What parser (default, Babel-ESLint, etc.) are you using?**
Babel-ESLint

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
	"parser": "babel-eslint",
	"parserOptions": {
		"sourceType": "module",
		"ecmaVersion": 9,
		"ecmaFeatures": {
			"jsx": true
		}
	},
	"extends": [
		"eslint:recommended",
		"plugin:flowtype/recommended",
		"google",
		"prettier"
	],
	"env": {
		"amd": true,
		"browser": true,
		"es6": true,
		"mocha": true,
		"worker": true,
		"node": true
	},
	"globals": {
		"Inferno": true,
		"sinon": true
	},
	"plugins": ["mocha", "babel", "inferno", "flowtype"],
	"rules": {
		"no-invalid-this": 0,
		"babel/no-invalid-this": 2,
		"require-jsdoc": 0,
		"valid-jsdoc": 0,
		"linebreak-style": [2, "windows"],
		"inferno/jsx-uses-vars": "error",
		"require-atomic-updates": 0
	},
	"overrides": [
		{
			"files": ["tests/**/*.js"],
			"rules": {
				"camelcase": 0,
				"new-cap": 0,
				"no-invalid-this": 0,
				"babel/no-invalid-this": 0,
				"no-undef": 0,
				"no-unused-vars": 0,
				"prefer-rest-params": 0,
				"prefer-spread": 0,
				"mocha/no-exclusive-tests": "error"
			}
		}
	]
}
```

</details>

```bash
eslint --format codeframe .
```

I have the following structure:
```
- src
 - dir
 - legacyDir
   - dir1
   - dir2
     - file1.js
     - file2.js
```
And the following `.eslintignore` in the directory `src`:
```
legacyDir/
!legacyDir/dir2/file1.js
```

**What did you expect to happen?**
I wanted to fix legacy files step by step and added `!legacyDir/dir2/file1.js` to don't ignore one file from the whole directory `legacyDir`.

**What actually happened? Please include the actual, raw output from ESLint.**
Eslint ignores `!legacyDir/dir2/file1.js`.

I tried to debug eslint a little and looks like it happens because directory `legacyDir/dir2` detected as ignored and no internal files analyzed in this case.

![image](https://user-images.githubusercontent.com/37663273/75676831-1ded5200-5c9b-11ea-83a4-66197a822d9b.png)

**What I'm do wrong?**

**Comment by mdjermanovic** [comment]
Hi @abaksha-sc, thanks for the issue!

This works like in the [gitignore specification](https://git-scm.com/docs/gitignore). As you noticed in the title, it isn't possible to unignore a file if its parent (or any ancestor) directory is ignored, so the patterns have to be written to ignore the content.

Can you please try with the following `.eslintignore`:

```
legacyDir/**
!legacyDir/dir2/
!legacyDir/dir2/file1.js
```

**Comment by abaksha-sc** [comment]
@mdjermanovic , it works! Thank you!

**Comment by abaksha-sc** [comment]
Closed because it's an expected behavior (even when it's not obvious at least for me)
